node/root_peers.js

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var inherits = require('util').inherits;
+var extend = require('xtend');
+
+var TChannelPeersBase = require('./peers_base.js');
+var TChannelPeer = require('./peer');
+var TChannelSelfPeer = require('./self_peer');
+
+function TChannelRootPeers(channel, options) {
+    if (!(this instanceof TChannelRootPeers)) {
+        return new TChannelRootPeers(channel, options);
+    }
+
+    var self = this;
+    TChannelPeersBase.call(self, channel, options);
+
+    self.allocPeerEvent = self.defineEvent('allocPeer');
+    self.peerOptions = self.options.peerOptions || {};
+    self.selfPeer = null;
+}
+
+inherits(TChannelRootPeers, TChannelPeersBase);
+
+TChannelRootPeers.prototype.close = function close(callback) {
+    var self = this;
+
+    var peers = self.values();
+    if (self.selfPeer) {
```

Comment:
sure, let's punt this to future work, changing enough in this diff as is, and this was the easiest preservation of old logic, so keeping it for now.